scipy/stats/_anova_oneway.py

```diff
@@ -0,0 +1,405 @@
+
+from dataclasses import dataclass
+from collections import namedtuple
+import numpy as np
+from scipy import special
+from scipy.special import stdtrit
+from ._anova_util import (_make_table, _encode, _proj)
+
+
+__all__ = [
+    'AnovaOnewayResult',
+    'oneway', 'oneway_from_labels_values',
+]
+
+
+ConfidenceInterval = namedtuple('ConfidenceInterval', ['low', 'high'])
+Mean = namedtuple('Mean', ['mean', 'se', 'ci'])
+
+
+def _fmt(value):
+    return np.array2string(np.array(value))
+
+
+def _afmt(x):
+    # This code assumes x is a 1d sequence of floats or ints
+    s = [_fmt(t) for t in x]
+    s = [t + ('.' if '.' not in t else '') for t in s]
+    intpart, fracpart = zip(*[t.split('.') for t in s])
+    intwidth = max(len(t) for t in intpart)
+    fracwidth = max(len(t) for t in fracpart)
+    jintpart = [t.rjust(intwidth) for t in intpart]
+    jfracpart = [t.ljust(fracwidth) for t in fracpart]
+    js = ['.'.join(parts) for parts in zip(jintpart, jfracpart)]
+    return js
+
+
+@dataclass
+class AnovaOnewayResult:
+
+    mean: float
+    group_means: np.ndarray
+    group_sizes: np.ndarray
+    SSb: float
+    SSw: float
+    DFb: int
+    DFw: int
+    MSb: float
+    MSw: float
+    F: float
+    pvalue: float
+    levels: np.ndarray
+
+    def __str__(self):
+        vartot = self.SSb + self.SSw
+        dftot = self.DFb + self.DFw
+
+        header = "ANOVA one-way"
+        rowlabels = ['Between groups', 'Within groups', 'Total']
+        return _make_table(header, rowlabels,
+                           [self.SSb, self.SSw, vartot],
+                           [self.DFb, self.DFw, dftot],
+                           [self.MSb, self.MSw],
+                           [self.F],
+                           [self.pvalue])
+
+    def means_ci(self, confidence_level=0.95):
+        """
+        Confidence intervals for the means of the groups.
+        """
+        means = self.group_means
+        ngroups = len(means)
+        t = stdtrit(self.DFw, 0.5 + 0.5*confidence_level)
+        cis = []
+        for i in range(ngroups):
+            se = np.sqrt(self.MSw * (1/self.group_sizes[i]))
+            delta = t*se
+            cis.append(Mean(mean=means[i],
+                            se=se,
+                            ci=ConfidenceInterval(low=means[i] - delta,
+                                                  high=means[i] + delta)))
+        return cis
+
+    def deltas(self, confidence_level=0.95):
```

Comment:
I wouldn't add those.
They will be misleading. Users should call tukey-hsd instead